chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/certifiedProduct/CertifiedProduct2014Validator.java

```diff
@@ -272,16 +272,16 @@ protected void validateDemographics(CertifiedProductSearchDetails product) {
 		for(CertificationResult cert : product.getCertificationResults()) {
 			if(cert.getTestToolsUsed() != null && cert.getTestToolsUsed().size() > 0) {
 				for(CertificationResultTestTool testTool : cert.getTestToolsUsed()) {
-					if(testTool.getTestToolId() == null) {
-						product.getErrorMessages().add("There was no test tool found matching '" + testTool.getTestToolName() + "' for certification " + cert.getNumber() + ".");
+					if(StringUtils.isEmpty(testTool.getTestToolName())) {
+						product.getErrorMessages().add("There was no test tool name found for certification " + cert.getNumber() + ".");
 					} else {
-						TestToolDTO tt = super.testToolDao.getById(testTool.getTestToolId());
+						TestToolDTO tt = super.testToolDao.getByName(testTool.getTestToolName());
 						if(tt != null && tt.isRetired() && super.icsCode.equals("0")) {
-							if(super.hasIcsConflict){
+							if(super.hasIcsConflict) {
 								product.getWarningMessages().add("Test Tool '" + testTool.getTestToolName() + "' can not be used for criteria '" + cert.getNumber() 
 								+ "', as it is a retired tool, and this Certified Product does not carry ICS.");
 							}
-							else {
+							else if(tt != null) {
 								product.getErrorMessages().add("Test Tool '" + testTool.getTestToolName() + "' can not be used for criteria '" + cert.getNumber() 
```

Comment:
This is not the right error message. should be something like "No test tool with testTool.getTestToolName() was found for criteria cert.getNumber()